src/main/scala/io.scala

```diff
@@ -151,4 +170,75 @@ case object io {
       treeMapToTaxTree(tree)
     }
 
+  def dumpTaxTreeToFiles(
+      tree: TaxTree,
+      dataFile: File,
+      shapeFile: File
+  ): FileError + (File, File) = {
+
+    val format = defaultFormat
+
+    val serialization = treeIO.serializeTree(tree, format)
+    // data and shape are numberedLines, we need to map them to
+    // their first element to dump them to a file
+    val data  = serialization.data.map { _._1 }
+    val shape = serialization.shape.map { _._1 }
+
+    val dataResult = write.linesToFile(dataFile)(data)
+
+    dataResult.flatMap { dataFile =>
+      val shapeResult = write.linesToFile(shapeFile)(shape)
+      shapeResult.map { shapeFile =>
+        (dataFile, shapeFile)
+      }
+    }
+  }
+
+  def readTaxTreeFromFiles(
+      dataFile: File,
+      shapeFile: File
+  ): FileError + (SerializationError + TaxTree) = {
```

Comment:
I like to group the errors in the left side, as in `(FileError + SerializationError) + TaxTree`, just to be able to directly map the either and work with the result. Although this is just a preference, we should agree to always implement these values in one or the other way. Just saying it here to open a discussion :) What do you prefer?

Reply:
You are right, your option is better. I will update it before merging the PR

Reply:
Oh, that was an easy discussion!